test/e2e/upgrades/upgrade.go

```diff
@@ -41,6 +41,9 @@ const (
 
 // Test is an interface for upgrade tests.
 type Test interface {
+	// Name should return a test name sans spaces.
+	Name() string
```

Comment:
Reflection would make this obtuse. I'd prefer it to be clear.

What do you mean by "framework array"? The array of tests at the top? The map of frameworks?